src/amm/mod.rs

```diff
@@ -19,46 +21,53 @@ use self::{erc_4626::ERC4626Vault, uniswap_v2::UniswapV2Pool, uniswap_v3::Uniswa
 #[async_trait]
 pub trait AutomatedMarketMaker {
     /// Returns the address of the AMM.
-    fn address(&self) -> H160;
+    fn address(&self) -> Address;
 
     /// Syncs the AMM data on chain via batched static calls.
-    async fn sync<M: Middleware>(&mut self, middleware: Arc<M>) -> Result<(), AMMError<M>>;
+    async fn sync<T: Transport + Clone, P: Provider<T, AnyNetwork>>(
+        &mut self,
+        provider: Arc<P>,
+    ) -> Result<(), AMMError>;
 
     /// Returns the vector of event signatures subscribed to when syncing the AMM.
-    fn sync_on_event_signatures(&self) -> Vec<H256>;
+    fn sync_on_event_signatures(&self) -> Vec<B256>;
 
     /// Returns a vector of tokens in the AMM.
-    fn tokens(&self) -> Vec<H160>;
+    fn tokens(&self) -> Vec<Address>;
 
     /// Calculates a f64 representation of base token price in the AMM.
-    fn calculate_price(&self, base_token: H160) -> Result<f64, ArithmeticError>;
+    fn calculate_price(&self, base_token: Address) -> Result<f64, ArithmeticError>;
 
     /// Updates the AMM data from a log.
     fn sync_from_log(&mut self, log: Log) -> Result<(), EventLogError>;
 
     /// Populates the AMM data via batched static calls.
-    async fn populate_data<M: Middleware>(
+    async fn populate_data<T: Transport + Clone, P: Provider<T, AnyNetwork>>(
```

Comment:
Same thing here, can we use `where` clauses for readability? Lets apply this across the AMM trait for instances like this.